archunit/src/test/java/com/tngtech/archunit/lang/syntax/elements/CodeUnitsShouldTest.java

```diff
@@ -277,6 +285,345 @@ public void throwable_type_predicates(ArchRule rule, Collection<String> expected
         assertThat(actualMembers).containsOnlyElementsOf(expectedMembers);
     }
 
+    @DataProvider
+    public static Object[][] restricted_constructor_calls_by_classes_rules() {
+        return $$(
+                $(constructors().should(onlyBeCalledByClassesThat(equivalentTo(ClassCorrectlyCallingMethodAndConstructor.class)))),
```

Comment:
Why not use `JavaClass.Predicates.belongToAnyOf(..)` as the predicate, instead of `equivalentTo`? Then it would be consistent with the fluent API case two lines below...